### PR TITLE
[Feature][Iceberg] Support custom delete sql for sink savemode

### PR DESCRIPTION
**Pull request by hailin0**
### Purpose of this pull request

[Iceberg] Support custom delete sql for sink savemode

### Does this PR introduce _any_ user-facing change?

Add new config

### How was this patch tested?

Added

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: docs/en/connector-v2/sink/Iceberg.md [hunk 1]
```diff
@@ -76,6 +76,7 @@ libfb303-xxx.jar
...
 | data_save_mode                         | Enum    | no       | APPEND_DATA                  | the data save mode, please refer to `data_save_mode` below                                                                                                                                                                                                                                                                |
+| data_save_mode_custom_sql              | string  | no       | -                            | Custom `delete` data sql for data save mode. e.g: `delete from ... where ...`                                                                                                                                                                                                                                             |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/sink/Iceberg.md]:
```suggestion
| custom_sql              | string  | no       | -                            | Custom `delete` data sql for data save mode. e.g: `delete from ... where ...`                                                                                                                                                                                                                                             |
```
Can we just use `custom_sql` seem as https://seatunnel.apache.org/docs/2.3.8/connector-v2/sink/Jdbc#custom_sql-string

Comment (hailin0) [reply docs/en/connector-v2/sink/Iceberg.md]: fixed

File: docs/en/connector-v2/sink/Iceberg.md [hunk 2]
```diff
@@ -76,6 +76,7 @@ libfb303-xxx.jar
...
 | data_save_mode                         | Enum    | no       | APPEND_DATA                  | the data save mode, please refer to `data_save_mode` below                                                                                                                                                                                                                                                                |
+| custom_sql                             | string  | no       | -                            | Custom `delete` data sql for data save mode. e.g: `delete from ... where ...`                                                                                                                                                                                                                                             |
```
Comment (Hisoka-X) [inline docs/en/connector-v2/sink/Iceberg.md:79]: Basic lgtm. But I'm a little worried, is that the all connector of not support sql we would use our own sql style to support delete by SQL?

Comment (hailin0) [reply docs/en/connector-v2/sink/Iceberg.md:79]: No, iceberg has a data deletion API that is very similar to the SQL language style, which is the premise for implementing this function.

Comment (Hisoka-X) [reply docs/en/connector-v2/sink/Iceberg.md:79]: provide some link into doc?

Comment (hailin0) [reply docs/en/connector-v2/sink/Iceberg.md:79]: Please look at this method

https://github.com/apache/iceberg/blob/430ebff8ea4f947a8cef1ad181b19ed5157ef26b/api/src/main/java/org/apache/iceberg/expressions/Expressions.java#L32
https://github.com/apache/iceberg/blob/430ebff8ea4f947a8cef1ad181b19ed5157ef26b/docs/docs/api.md?plain=1#L136

**Review by Hisoka-X — APPROVED** [review]

**Review by davidzollo — APPROVED** [review]
+1
LGTM

**Pull request merged**